tests/test_velocity_verlet_integrator.py

```diff
@@ -0,0 +1,153 @@
+from functools import partial
+
+import numpy as np
+
+from timemachine.integrator import VelocityVerletIntegrator as ReferenceVelocityVerlet
+from timemachine.lib import VelocityVerletIntegrator, custom_ops
+from timemachine.lib.potentials import SummedPotential
+from timemachine.testsystems.relative import hif2a_ligand_pair
+
+
+def setup_velocity_verlet(bps, x0, box, dt, masses):
+    integrator = VelocityVerletIntegrator(dt, masses)
+    # return integrator impl to avoid deallocating
+    intg = integrator.impl()
+    context = custom_ops.Context(x0, np.zeros_like(x0), box, intg, bps)
+    return intg, context
+
+
+def assert_reversible(x0, v0, update_fxn, lambdas, atol=1e-10):
+    """Define a fxn self_inverse as composition of flip_velocities and update_fxn,
+    then assert that
+    * self_inverse is its own inverse
+    * self_inverse is not trivial (aka not the identity function)
+    """
+
+    def self_inverse(x, v, lamb_sched):
+        """integrate forward in time, flip v
+        (expected to be an "involution" i.e. its own inverse)"""
+        x_next, v_next = update_fxn(x, v, lamb_sched)
+        return x_next, -v_next, lamb_sched[::-1]
+
+    # assert "self_inverse" is really its own inverse
+    x1, v1, rev_lamb = self_inverse(x0, v0, lambdas)
+    x0_, v0_, _ = self_inverse(x1, v1, rev_lamb)
+
+    np.testing.assert_almost_equal(x0_, x0, decimal=10)
+    np.testing.assert_almost_equal(v0_, v0, decimal=10)
```

Comment:
nit: from the [docs](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html):

> It is recommended to use one of [assert_allclose](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose), [assert_array_almost_equal_nulp](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_almost_equal_nulp.html#numpy.testing.assert_array_almost_equal_nulp) or [assert_array_max_ulp](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_max_ulp.html#numpy.testing.assert_array_max_ulp) instead of this function for more consistent floating point comparisons.

Should we standardize on `assert_allclose` since we're using it elsewhere?